dask/utils.py

```diff
@@ -924,3 +924,73 @@ def natural_sort_key(s):
     """
     return [int(part) if part.isdigit() else part
             for part in re.split('(\d+)', s)]
+
+
+def factors(n):
+    """ Return the factors of an integer
+
+    https://stackoverflow.com/a/6800214/616616
+    """
+    seq = ([i, n // i] for i in range(1, int(pow(n, 0.5) + 1)) if n % i == 0)
+    return set(functools.reduce(list.__add__, seq))
+
+
+def parse_bytes(s):
+    """ Parse byte string to numbers
+
+    >>> parse_bytes('100')
+    100
+    >>> parse_bytes('100 MB')
+    100000000
+    >>> parse_bytes('100M')
+    100000000
+    >>> parse_bytes('5kB')
+    5000
+    >>> parse_bytes('5.4 kB')
+    5400
+    >>> parse_bytes('1kiB')
+    1024
+    >>> parse_bytes('1e6')
+    1000000
+    >>> parse_bytes('1e6 kB')
+    1000000000
+    >>> parse_bytes('MB')
+    1000000
+    """
+    s = s.replace(' ', '')
+    if not s[0].isdigit():
+        s = '1' + s
+
+    for i in range(len(s) - 1, -1, -1):
+        if not s[i].isalpha():
+            break
+    index = i + 1
+
+    prefix = s[:index]
+    suffix = s[index:]
+
+    n = float(prefix)
+
+    multiplier = byte_sizes[suffix.lower()]
```

Comment:
This would be a good point to catch `ValueError` (from `float`) and `KeyError` (from `bytes_sizes`) to report an informative error message to the user, e.g., `ValueError: could not interpret size as a number of bytes: 'your invalid string'`.

Reply:
I've added informative errors around both possible failures.